src/gc-debug.c

```diff
@@ -579,27 +579,25 @@ static void gc_scrub_task(jl_task_t *ta)
     int16_t tid = ta->tid;
     jl_ptls_t ptls = jl_get_ptls_states();
     jl_ptls_t ptls2 = jl_all_tls_states[tid];
+
+    char *low;
+    char *high;
+    if (ta->copy_stack && ta == ptls2->current_task) {
+        low  = (char*)ptls2->stackbase;
+        high = (char*)ptls2->stackbase + ptls2->stacksize;
+    }
+    else if (ta->stkbuf) {
+        low  = (char*)ta->stkbuf;
+        high = (char*)ta->stkbuf + ta->bufsz;
+    }
+    else
+        return;
+
     if (ptls == ptls2 && ta == ptls2->current_task) {
         // scan up to current `sp` for current thread and task
         char *low = (char*)jl_get_frame_addr();
```

Comment:
in the future, we could probably store the precise answer during task switch too, and at the start of gc for most threads (that just leaves still-running threads in gc-safe regions)